undi/undi_analysis.py

```diff
@@ -0,0 +1,284 @@
+import sys, json, argparse
+from itertools import product
+import numpy as np
+
+from ase import Atoms
+from ase.io import read
+from ase.neighborlist import neighbor_list
+
+from undi import MuonNuclearInteraction
+from .isotopes import Element
+
+angtom = 1e-10
+
+def gen_neighbouring_atomic_structure(atoms, isotopes, spins, hdim_max):
+
+    ai,aj,d,D = neighbor_list('ijdD',atoms, 10.) # very large cutoff to get all possible interactions.
+                                              # Actual selection is done below
+
+    try:
+        EFGs = atoms.get_array('efg') # EFG should be stored in the atoms object. Done via read_magres_file() function (see below).
+    except:
+        EFGs = np.array([0])
+        
+    # sort
+    srt = np.argsort(d)
+    ai = ai[srt]
+    aj = aj[srt]
+    d = d[srt]
+    D = D[srt]
+
+    data = []
+    muon_pos = np.array([0,0,0])
+    hdim = 2 # the muon Hilbert space
+
+    for i in range(len(D)):
+
+        if not (ai[i] == len(atoms)-1):
+            continue
+
+        atom_symbol = atoms[aj[i]].symbol
+
+        isotope_spin = spins[atom_symbol]
```

Comment:
In testing Si2H6 case, I get also atom_symbol='H', raising an error. Is this expected?

Reply:
yes it is... I see that this fails just for Si (I also used Cobalt, and it works fine)